### PR TITLE
[RPA-1266] Add process collection path breadcrumbs

### PR DESCRIPTION
**Pull request by Aeri4a**
## Description
New look without 'Home' label
image

image

image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java [hunk 1]
```diff
@@ -55,5 +55,20 @@ public interface ProcessCollectionRepository extends JpaRepository<ProcessCollec
...
+            "UNION " +
+            "SELECT pc.*, bc.lvl + 1 FROM bread_crumbs bc " +
+            "JOIN process_collection pc ON pc.id = bc.parent_id " +
+        ") " +
+        "SELECT bc.id, bc.name, bc.description, bc.created, bc.updated, bc.created_by, bc.is_public, bc.parent_id " +
+            "FROM bread_crumbs bc " +
+            "ORDER BY bc.lvl DESC",
+        nativeQuery = true
+    )
+    List<ProcessCollection> findAllCollectionHierarchy(UUID id);
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java]: what do you think about the name getAllAncestors?

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import styled from 'styled-components';
+
+export const StyledLink = styled(Link)`
+    text-decoration: none;
+`;
+
+export const HomeBox = styled(Box)`
+    display: flex;
+    min-height: 24px;
+    align-items: center;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.styles.tsx]: switch the order of these two

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.styles.tsx]: lines

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+
+import homeIcon from '#public/images/icons/home.svg';
+import { CollectionBreadcrumb } from '#src-app/hooks/useProcessCollection';
+import useTranslations from '#src-app/hooks/useTranslations';
+
+import { HomeBox, StyledLink } from './ProcessCollectionPath.styles';
+
+const ACTIVE_ICON_COLOR_FILTER = 'invert(72%) sepia(41%) saturate(3394%) hue-rotate(356deg) brightness(101%) contrast(94%)';
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.tsx]: yesterday I've found out that the lock is mui icon 
use 
`import LockOutlinedIcon from '@mui/icons-material/LockOutlined';`
changing the color should be easy from now on

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                        <Image
+                            src={homeIcon}
+                            alt='home icon'
+                            width='20'
+                            height='20'
+                            style={isRootFolder ? { filter: ACTIVE_ICON_COLOR_FILTER } : {}}
+                        />
+                        <Typography
+                            variant='body2'
+                            color={isRootFolder ? 'secondary' : 'black'}
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.tsx]: imo 'black' should be replaced with some color from theme (use useTheme() hook)

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                        }
+                    }}
+                >
+                    <HomeBox>
+                        <Image
+                            src={homeIcon}
+                            alt='home icon'
+                            width='20'
+                            height='20'
+                            style={isRootFolder ? { filter: ACTIVE_ICON_COLOR_FILTER } : {}}
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessCollectionTile/ProcessCollectionPath/ProcessCollectionPath.tsx]: it's overwriting the html styles - we're not supposed to do that
but if you use Mui icon (there's a comment above) problem solved - you can use 
`<HomeIcon color="targetColor" />`
or
`<HomeIcon sx={{ color: pink[500] }} />`

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.extraReducers.ts [hunk 6]
```diff
@@ -16,6 +16,11 @@ const buildProcessCollectionExtraReducers = (builder: ActionReducerMapBuilder<Pr
         })
         .addCase(createOne.rejected, (state) => {
             state.childrenCollections.isLoading = false;
+        })
+
+        // GET PATH
+        .addCase(getPath.fulfilled, (state, action) => {
+            state.pathCollections = [...action.payload];
         });
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.extraReducers.ts]: add cases for rejected and pending (imo add 'loading' prop to redux state and put skeleton loader in breadcrumbs place unless it's loaded and show the snackbar 'unexpected error - could not load navigation' or sth like that if the breadcrumbs were rejected)

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.slice.ts [hunk 7]
```diff
@@ -15,7 +15,8 @@ const initialState: ProcessCollectionState = {
     childrenCollections: {
         isLoading: false,
         list: [],
-    }
+    },
+    pathCollections: []
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.slice.ts]: i like active.collectionHierarchy or active.ancestors more

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.thunks.ts [hunk 8]
```diff
@@ -2,8 +2,15 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 import axios from 'axios';
 import { ProcessCollection } from 'runbotics-common';
 
+import URLBuilder from '#src-app/utils/URLBuilder';
+
 import { CollectionCreateParams } from './ProcessCollection.types';
 
+const buildCollectionPathURL = (collectionId) =>  URLBuilder
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.thunks.ts]: 2x space ;p

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.thunks.ts [hunk 9]
```diff
@@ -2,8 +2,15 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
...
+    .url('/api/process-collection-path')
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessCollection/ProcessCollection.thunks.ts]: process-collection-hierarchy or process-collection/current/ancestors

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/collections/general.ts [hunk 10]
```diff
@@ -22,6 +22,7 @@ const general: typeof englishGeneral = {
     'Process.Collection.Tile.MenuItem.Edit': 'Edytuj',
     'Process.Collection.Navigation.Processes.Label': 'Procesy',
     'Process.Collection.Navigation.Collections.Label': 'Zbiory',
+    'Process.Collection.Path.Home': 'Dom',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/collections/general.ts]: ask on RPA Offtop - what's the best translation - i'm not sure neither about 'dom', nor 'strona główna'

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBrowseView/Header.tsx [hunk 11]
```diff
@@ -57,18 +57,20 @@ const Header: FC<HeaderProps> = ({ className, ...rest }) => {
...
     const onTabChange = (event: ChangeEvent<HTMLInputElement>, value: ProcessesTabs) => {
         if (value === ProcessesTabs.COLLECTIONS) {
-            router.replace({ 
+            router.replace({
                 pathname:  '/app/processes/collections',
                 query: {
-                    collection: collectionId,
-                    pageSize: pageSize, 
+                    collectionId,
+                    locale: router.locale,
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBrowseView/Header.tsx]: as far as I'm concerned we don't need to pass the locale - the useTranslations hook takes care of that

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBrowseView/Header.tsx [hunk 12]
```diff
@@ -57,18 +57,20 @@ const Header: FC<HeaderProps> = ({ className, ...rest }) => {
...
                     page: currentPage
                 }
-            }); 
+            });
         } else {
-            router.replace({ 
+            router.replace({
                 pathname:  '/app/processes',
                 query: {
+                    locale: router.locale,
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBrowseView/Header.tsx]: same

**Review by star-in-the-jar — DISMISSED** [review]

File: runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java (hunk 1 above)
Comment (Aeri4a) [reply runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java]: Sure, I will go with this concept.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/collections/general.ts (hunk 10 above)
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/collections/general.ts]: Not sure if we should directly translate it, sounds funny

File: runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java [hunk 13]
```diff
@@ -55,5 +55,20 @@ public interface ProcessCollectionRepository extends JpaRepository<ProcessCollec
...
+            "WHERE pc.id = ?1 " +
+            "UNION " +
+            "SELECT pc.*, bc.lvl + 1 FROM bread_crumbs bc " +
+            "JOIN process_collection pc ON pc.id = bc.parent_id " +
+        ") " +
+        "SELECT bc.id, bc.name, bc.description, bc.created, bc.updated, bc.created_by, bc.is_public, bc.parent_id " +
+            "FROM bread_crumbs bc " +
+            "ORDER BY bc.lvl DESC",
+        nativeQuery = true
+    )
```
Comment (preludium) [inline runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java:71]: what is example output of it?

**Review by preludium — APPROVED** [review]

File: runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java (hunk 13 above)
Comment (Aeri4a) [reply runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java:71]: A list of collections in path (ancestors) sorted by level of recursion, first element (with index 0) will have parent id as null, so in it is in root collection, next element will be a child of this and so on. Element with the highest index will be our active collection.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/collections/general.ts (hunk 10 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/collections/general.ts]: So, we decided to stick with just icon, without label.

**Review by preludium — APPROVED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessCollection.ts [hunk 14]
```diff
@@ -1,18 +1,54 @@
...
 const getCollectionOwner = (collectionId: CollectionId) => true; // todo
 
 const useProcessCollection = () => {
-    const collectionId = useSearchParams().get(COLLECTION_ID_PARAM) ?? ROOT_PROCESS_COLLECTION_ID;
-    // const breadcrumbs = getBreadcrumbs(collectionId); // todo
+    const dispatch = useDispatch();
+    const { translate } = useTranslations();
+    const { enqueueSnackbar } = useSnackbar();
+    const collectionId: string = useSearchParams().get(COLLECTION_ID_PARAM) ?? ROOT_PROCESS_COLLECTION_ID;
+    const { active: { ancestors: { list: collectionAncestors } } } = useSelector(processCollectionSelector);
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessCollection.ts:31]: interesting, I had no idea its possible 😃

**Review by star-in-the-jar — APPROVED** [review]
looks solid

**Pull request merged**